src/pages/home/sidebar/SidebarLinks.js

```diff
@@ -158,23 +159,34 @@ class SidebarLinks extends React.Component {
         this.state = {
             currentlyViewedReportID: props.currentlyViewedReportID,
             orderedReports: [],
+            priorityMode: props.priorityMode,
             unreadReports: SidebarLinks.getUnreadReports(props.reports || {}),
         };
     }
 
     static getDerivedStateFromProps(nextProps, prevState) {
         const shouldReorder = SidebarLinks.shouldReorder(nextProps, prevState.orderedReports, prevState.currentlyViewedReportID, prevState.unreadReports);
+        const switchingPriorityModes = nextProps.priorityMode !== prevState.priorityMode;
```

Comment:
Can we compare `this.props.priorityMode` with `nextProps.priorityMode` and leave `state` out of the equation entirely?

Reply:
this.props isn't available within `getDerivedStateFromProps` for some reason. 

Out of curiosity, why do you want to leave state out of the equation? Bc we are not using `state.priorityMode` anywhere else?

Reply:
>Bc we are not using state.priorityMode anywhere else

That, and we're just setting whatever is in props directly to state which _usually_ means it does not need to be there. If we need to track the updates of a prop then we can just look at the prop.

Anyway, I think we are basically stuck at this point and would need to refactor the entire component to fix this. I'm remembering now that this is a _static_ method and can't access the component instance at all.

There are some good explanations here that go into why we should not use this here (but hopefully the explanation I just gave also makes sense).

https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html

Reply:
I would actually be super interested in refactoring the entire component

However, should I scale back some of the changes then to get somethign merged now? Right now the undefineds are causing a crash, or requires a hard refresh

Reply:
Yeah! I created this issue [here](https://github.com/Expensify/App/issues/7948) - feel free to steal it from me. 

But agree we should just fix this for now :D